ci/check-version-file.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+set -euo pipefail
+
+protected_branches=(release preview)
```

Comment:
What about master?

Reply:
To document: Master is intended to be on a floating engine version, release and preview are protected, i.e. required to have a pinned engine version.